### PR TITLE
DM-7150: Configure a default log for app unit tests

### PR DESCRIPTION
**Pull request by hchiang2**

### CODE REVIEW

File: python/lsst/utils/tests.py [hunk 1]
```diff
@@ -71,6 +72,13 @@ def init():
         memId0 = dafBase.Citizen_getNextMemId()  # used by MemoryTestCase
     # Reset the list of open files
     open_files = _get_open_files()
+    lsst.log.configure_prop("""
```
Comment (timj) [inline python/lsst/utils/tests.py]: Can you add a comment before this line to explain what you are doing.

File: python/lsst/utils/tests.py [hunk 2]
```diff
@@ -71,6 +72,13 @@ def init():
         memId0 = dafBase.Citizen_getNextMemId()  # used by MemoryTestCase
```
Comment (timj) [inline python/lsst/utils/tests.py]: I know this is not your fault, but please add a docstring to this function to explain what it does. It didn't really matter when the only thing it did was set the watermark for the memory leak tester.

Comment (hchiang2) [inline python/lsst/utils/tests.py]: docstring added

File: python/lsst/utils/tests.py (hunk 1 above)
Comment (hchiang2) [inline python/lsst/utils/tests.py]: comment added

File: python/lsst/utils/tests.py [hunk 3]
```diff
@@ -65,12 +66,21 @@ def _get_open_files():
...
 
 def init():
+    """Initialize the memory tester and configure a default log."""
     global memId0
     global open_files
     if dafBase:
         memId0 = dafBase.Citizen_getNextMemId()  # used by MemoryTestCase
     # Reset the list of open files
     open_files = _get_open_files()
+    # Setup a default cononfiguration for log used in the test framework
```
Comment (timj) [inline python/lsst/utils/tests.py]: Typo "cononfiguration"

Comment (hchiang2) [inline python/lsst/utils/tests.py]: Thanks for catching it.

**Pull request merged**